pkg/kademlia/routing.go

```diff
@@ -149,6 +149,19 @@ func (rt *RoutingTable) GetBucketIds() (storage.Keys, error) {
 	return kbuckets, nil
 }
 
+// DumpNodes iterates through all nodes in the nodeBucketDB and marshals them to &pb.Nodes, then returns them
+func (rt *RoutingTable) DumpNodes() ([]*pb.Node, error) {
+	var nodes []*pb.Node
+	err := rt.iterate(storj.NodeID{}, func(newID storj.NodeID, protoNode []byte) error {
+		newNode := pb.Node{}
+		err := proto.Unmarshal(protoNode, &newNode)
+		nodes = append(nodes, &newNode)
+		return err
+	})
+
+	return nodes, Error.Wrap(err)
```

Comment:
I think its safer to return nil array when an error occurs.

Reply:
Same as above, using ErrorGroup here for this.